wrapdocker

```diff
@@ -136,53 +136,57 @@ if [ "${STORAGE_FS}" != "overlay" -o "${OVERLAY_WORKS}" == false ]; then
   mount -o loop "${STORAGE_FILE}" "${STORAGE_DIR}"
 fi
 
+DOCKER_DAEMON_ARGS="${DOCKER_DAEMON_ARGS:-} --storage-driver=${STORAGE_FS}"
+
 # If a pidfile is still around (for example after a container restart),
 # delete it so that docker can start.
 rm -rf /var/run/docker.pid
 
-# create docker0 bridge manually and attach it to the veth interface eth0
-brctl addbr docker0
-brctl addif docker0 eth0
-ip link set docker0 up
-
-# move ip to the bridge and restore routing via the old gateway
-IP_CIDR=$(ip addr show eth0 | grep -w inet | awk '{ print $2; }')
-IP=$(echo $IP_CIDR | sed 's,/.*,,')
-NETWORK_SIZE=$(echo $IP_CIDR | sed 's,.*/,,')
-DEFAULT_ROUTE=$(ip route | grep default | sed 's/eth0/docker0/')
-
-ip addr del $IP_CIDR dev eth0
-ip addr add $IP_CIDR dev docker0
-ip route add $DEFAULT_ROUTE
-
-# compute a network for the containers to live in
-# by adding DOCKER_NETWORK_OFFSET to the current IP and cutting off
-# non-network bits according to DOCKER_NETWORK_SIZE
-DOCKER_NETWORK_SIZE=${DOCKER_NETWORK_SIZE:-24}
-DOCKER_NETWORK_OFFSET=${DOCKER_NETWORK_OFFSET:-0.0.1.0}
-NETWORK=$(ip route | grep docker0 | grep -v default | sed 's,/.*,,')
-
-IFS=. read -r i1 i2 i3 i4 <<< $IP
-IFS=. read -r n1 n2 n3 n4 <<< $NETWORK
-IFS=. read -r o1 o2 o3 o4 <<< $DOCKER_NETWORK_OFFSET
-IFS=. read -r w1 w2 w3 w4 <<< $(ipcalc $IP_CIDR | grep Wildcard | awk '{print $2;}')
-
-IP_PLUS_OFFSET=$(printf "%d.%d.%d.%d\n" \
-  "$(( n1 + ((i1 - n1 + o1) & w1) ))" \
-  "$(( n2 + ((i2 - n2 + o2) & w2) ))" \
-  "$(( n3 + ((i3 - n3 + o3) & w3) ))" \
-  "$(( n4 + ((i4 - n4 + o4) & w4) ))")
-
-FIXED_CIDR=$(ipcalc $IP_PLUS_OFFSET/$DOCKER_NETWORK_SIZE | grep Network | awk '{print $2;}')
-echo "Using network $FIXED_CIDR for docker containers"
+if [ "${DOCKER_NETWORK_BRIDGE:-false}" == "true" ]; then
```

Comment:
ROUTABLE_CONTAINER_IPS would be a better name. There is always a docker bridge. The only difference here is that we set it up in a special way.